eogrow/core/area/base.py

```diff
@@ -248,3 +249,92 @@ def _add_bbox_column(grid: List[gpd.GeoDataFrame]) -> None:
         for bbox_df in grid:
             crs = CRS(bbox_df.crs)
             bbox_df["BBOX"] = bbox_df.geometry.apply(lambda geo: BBox(geo.bounds, crs))  # noqa B023
+
+
+class BaseAreaManager(EOGrowObject, metaclass=ABCMeta):
+    """A manager for the AOI and how it is split into EOPatches"""
+
+    NAME_COLUMN = "eopatch_name"
+
+    class Schema(ManagerSchema):
+        ...
+
+    config: Schema
+
+    def __init__(self, config: Schema, storage: StorageManager):
+        """
+        :param config: A configuration file
+        :param storage: An instance of StorageManager class
+        """
+        super().__init__(config)
+
+        self.storage = storage
+
+    @abstractmethod
+    def get_area_geometry(self, *, crs: CRS = CRS.WGS84) -> Geometry:
+        """Provides a single geometry object of entire AOI"""
+
+    def get_grid(self) -> Dict[CRS, gpd.GeoDataFrame]:
+        """Provides a grid of bounding boxes which divide the AOI. Uses caching to avoid recalculations.
+
+        The grid is split into different CRS zones.
+
+        The `geometry` column serves as BBox geometry definition. The `geom.bounds` is taken as the definition.
+
+        The EOPatch names are stored in a column with identifier `self.NAME_COLUMN`.
+
+        :return: A dictionary of GeoDataFrames that defines how the area is split into EOPatches.
+        """
+        grid_path = fs.path.combine(self.storage.get_cache_folder(), self.get_grid_cache_filename())
+
+        if self.storage.filesystem.exists(grid_path):
+            return self._load_grid(grid_path)
+
+        grid = self._create_grid()
+        self._save_grid(grid, grid_path)
+        return grid
+
+    @abstractmethod
+    def _create_grid(self) -> Dict[CRS, gpd.GeoDataFrame]:
+        """Defines a new grid, which encodes how the area is split into EOPatches.
+
+        The grid is split into different CRS zones.
+
+        The `geometry` column serves as BBox geometry definition, with the implication that for a geometry `geom`
+        the BBox is defined as `geom.bounds`.
+
+        The EOPatch names are stored in a column with identifier `eopatch_name`.
+        """
```

Comment:
Not sure what happened here, function has almost the same docstring and nothing else (no content, no return)

Perhaps you're setting things up in advance?

Reply:
Didn't understand completely the abstract classes. Learned something new. Resolving